### PR TITLE
Further tests and postsubmit for Upgrades Operator

### PR DESCRIPTION
**Pull request by irinamihai**
Description:
- add new ci-job and integration tests
- add new image and generate postsubmit yaml file

### CODE REVIEW

File: core-services/image-mirroring/openshift-kni/mapping_openshift-kni_quay [hunk 1]
```diff
@@ -35,3 +35,5 @@ registry.ci.openshift.org/ocp-kni/4.9:cnf-tests quay.io/openshift-kni/cnf-tests:
 registry.ci.openshift.org/ocp-kni/4.9:dpdk quay.io/openshift-kni/dpdk:4.9 quay.io/openshift-kni/dpdk:4.9.0
 registry.ci.openshift.org/ocp-kni/4.10:cnf-tests quay.io/openshift-kni/cnf-tests:4.10 quay.io/openshift-kni/cnf-tests:4.10.0 quay.io/openshift-kni/cnf-tests:latest
 registry.ci.openshift.org/ocp-kni/4.10:dpdk quay.io/openshift-kni/dpdk:4.10 quay.io/openshift-kni/dpdk:4.10.0 quay.io/openshift-kni/dpdk:latest
+# cluster-group-upgrades-operator
+registry.ci.openshift.org/ocp-kni/cluster-group-upgrades-operator:cluster-group-upgrades-operator quay.io/openshift-kni/cluster-group-upgrades-operator:4.10-snapshot
```
Comment (fedepaol) [inline core-services/image-mirroring/openshift-kni/mapping_openshift-kni_quay:39]: it should be tag:image as the others.
Also, I don't see a repo created under quay

File: ci-operator/config/openshift-kni/cluster-group-upgrades-operator/openshift-kni-cluster-group-upgrades-operator-main.yaml [hunk 2]
```diff
@@ -33,6 +43,32 @@ tests:
...
+      OO_INSTALL_NAMESPACE: openshift-cluster-group-upgrades
+      OO_PACKAGE: cluster-group-upgrades-operator
+      OO_TARGET_NAMESPACES: '!all'
+    test:
+    - as: integration
+      cli: latest
+      commands: |
+        KUBECONFIG="${SHARED_DIR}/kubeconfig"
+        export KUBECONFIG
+        cp $(echo "${KUBECONFIG}") /go/src/github.com/openshift-kni/cluster-group-upgrades-operator/kubeconfig
```
Comment (fedepaol) [inline ci-operator/config/openshift-kni/cluster-group-upgrades-operator/openshift-kni-cluster-group-upgrades-operator-main.yaml:64]: Is having a local kubeconfig this required by the tests?

File: ci-operator/config/openshift-kni/cluster-group-upgrades-operator/openshift-kni-cluster-group-upgrades-operator-main.yaml [hunk 3]
```diff
@@ -1,25 +1,35 @@
 base_images:
   base:
-    name: "4.9"
+    name: ubi
```
Comment (fedepaol) [inline ci-operator/config/openshift-kni/cluster-group-upgrades-operator/openshift-kni-cluster-group-upgrades-operator-main.yaml:3]: The other examples I have at hand use 

```
  base:
    name: "4.10"
    namespace: ocp
    tag: base
```

Comment (irinamihai) [reply ci-operator/config/openshift-kni/cluster-group-upgrades-operator/openshift-kni-cluster-group-upgrades-operator-main.yaml:3]: I've mostly looked at node-label-operator and it's using the same base:
https://github.com/openshift/release/blob/master/ci-operator/config/openshift-kni/node-label-operator/openshift-kni-node-label-operator-main.yaml#L2. I also checked its Dockerfile: https://github.com/openshift-kni/node-label-operator/blob/main/Dockerfile#L2 which is the same as to what cluster-group-upgrades is using: https://github.com/openshift-kni/cluster-group-upgrades-operator/blob/main/Dockerfile#L2.

Please let me know what you think.

File: ci-operator/config/openshift-kni/cluster-group-upgrades-operator/openshift-kni-cluster-group-upgrades-operator-main.yaml (hunk 2 above)
Comment (irinamihai) [reply ci-operator/config/openshift-kni/cluster-group-upgrades-operator/openshift-kni-cluster-group-upgrades-operator-main.yaml:64]: I asked myself the same question here. I've tried it without this and the tests are failing. It seems to be a particularity of KUTTL. I've found another example here: https://github.com/openshift/release/blob/master/ci-operator/config/backube/volsync/backube-volsync-main.yaml#L116, the only other openshift/release example I could find that's using KUTTL in a similar manner.

File: core-services/image-mirroring/openshift-kni/mapping_openshift-kni_quay (hunk 1 above)
Comment (irinamihai) [reply core-services/image-mirroring/openshift-kni/mapping_openshift-kni_quay:39]: I see your point here, but looking at PAO, I noticed the format is different for the latest release:
registry.ci.openshift.org/ocp-kni/performance-addon-operator:performance-addon-operator quay.io/openshift-kni/performance-addon-operator:4.10-snapshot. Given the cluster-group-upgrades-operator is only supported starting with 4.10, I thought I could follow the same model.

Kindly let me know if you agree.

Will look into the repo being created under quay. Thank you!

Comment (fromanirh) [reply core-services/image-mirroring/openshift-kni/mapping_openshift-kni_quay:39]: Please note that PAO is not anymore a good example re: best practice. What PAO did is not necessarily the right thing to do in late 2021.

File: ci-operator/config/openshift-kni/cluster-group-upgrades-operator/openshift-kni-cluster-group-upgrades-operator-main.yaml (hunk 3 above)
Comment (fromanirh) [reply ci-operator/config/openshift-kni/cluster-group-upgrades-operator/openshift-kni-cluster-group-upgrades-operator-main.yaml:3]: From my POV, if using this image works for @irinamihai , I have no real concerns.

**Comment by fromanirh** [comment]
I share @fedepaol' s concern about copying KUBECONFIG around. This feels unusual at best. Do you have any plan in the future to stop doing that?

**Comment by irinamihai** [comment]
> I share @fedepaol' s concern about copying KUBECONFIG around. This feels unusual at best. Do you have any plan in the future to stop doing that?

I agree, it is unusual. I spent quite some time trying different approaches, but KUTTL doesn't want it other way. I suspect it might have to do with this issue that I've found: https://githubmemory.com/repo/kudobuilder/kuttl/issues/297. So even if IU don't copy the kubeconfig KUTTL will because of that issue. I also consulted with more knowledgeable persons on KUTTL (@aneeshkp , @mpryc ) and decided to use this approach for now.

My plan was to follow on that issue and update to the newest KUTTL version in the tests once it's fixed.
What I can do is to also add a TODO there to make sure we don't miss it in the future.

@fromanirh , kindly let me know what you think.

**Comment by fromanirh** [comment]
> > I share @fedepaol' s concern about copying KUBECONFIG around. This feels unusual at best. Do you have any plan in the future to stop doing that?
> 
> I agree, it is unusual. I spent quite some time trying different approaches, but KUTTL doesn't want it other way. I suspect it might have to do with this issue that I've found: https://githubmemory.com/repo/kudobuilder/kuttl/issues/297. So even if IU don't copy the kubeconfig KUTTL will because of that issue. I also consulted with more knowledgeable persons on KUTTL (@aneeshkp , @mpryc ) and decided to use this approach for now.
> 
> My plan was to follow on that issue and update to the newest KUTTL version in the tests once it's fixed. What I can do is to also add a TODO there to make sure we don't miss it in the future.
> 
> @fromanirh , kindly let me know what you think.

Works for me. I will take one final pass and approve if nothing else catches my eye. ETA is (late) Mon, Nov 8 (EU time).

**Comment by fromanirh** [comment]
@irinamihai to accelerate a bit more, could you please create an issue about removing the need to use kubeconfig in the relevant project and link it here? thanks!

**Review by fromanirh — APPROVED** [review]
/approve

**Pull request merged**